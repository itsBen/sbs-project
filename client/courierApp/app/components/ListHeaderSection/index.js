import React from 'react';
import { View, Text } from 'react-native';
import { Section } from 'react-native-tableview-simple';

export default ({ header, children, ...props }) => {
  return (
    <View style={styles.container}>
      <Section
        headerComponent={
          <View style={styles.headerContainer}>
            <Text style={styles.headerText}>{header}</Text>
          </View>
        }
        sectionPaddingTop={0}
        sectionTintColor="#F624591A"
        footerTextColor="#fff"
        {...props}
      >
        {children}
      </Section>
    </View>
  );
};

const styles = {
  container: {
    marginBottom: 10,
  },
  headerContainer: {
    backgroundColor: '#F624598C',
    padding: 10,
    borderWidth: 0,
  },
  headerText: {
    fontSize: 17,
    color: '#fff',
    fontWeight: '400',
  },
};
