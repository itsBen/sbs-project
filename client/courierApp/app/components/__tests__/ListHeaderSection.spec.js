/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ListHeaderSection from '../ListHeaderSection';

it('renders correctly', () => {
  const tree = renderer.create(<ListHeaderSection header="Header" />).toJSON();
  expect(tree).toMatchSnapshot();
});
