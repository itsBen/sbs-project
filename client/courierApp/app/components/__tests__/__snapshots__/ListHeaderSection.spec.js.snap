// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#F624591A",
          "paddingBottom": 15,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F624598C",
          "borderWidth": 0,
          "padding": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 17,
            "fontWeight": "400",
          }
        }
      >
        Header
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "backgroundColor": "#FFF",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 0.5,
            },
            Object {
              "backgroundColor": "#C8C7CC",
              "marginLeft": 0,
              "marginRight": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "backgroundColor": "#FFF",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 0.5,
            },
            Object {
              "backgroundColor": "#C8C7CC",
              "marginLeft": 0,
              "marginRight": 0,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
